density_scaler/src/settings.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{Error, Result};
+use config::{Config, Environment, File};
+use serde::Deserialize;
+use std::path::Path;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    /// RUST_LOG compatible settings string. Default to "density_scaler=debug"
+    #[serde(default = "default_log")]
+    pub log: String,
+    /// Follower settings
+    pub follower: node_follower::Settings,
+    /// Trigger every X minutes. Defaults to 30 mins.
+    #[serde(default = "default_trigger_interval")]
+    pub trigger: i64,
+}
+
+pub fn default_log() -> String {
+    "density_scaler=debug".to_string()
+}
+
+fn default_trigger_interval() -> i64 {
+    1800
+}
+
+impl Settings {
+    /// Load Settings from a given path. Settings are loaded from a given
+    /// optional path and can be overriden with environment variables.
+    ///
+    /// Environemnt overrides have the same name as the entries in the settings
+    /// file in uppercase and prefixed with "DENSITY_SCALER". For example
+    /// "DENSITY_SCALER_TRIGGER_INTERVAL" will override the trigger_interval
```

Comment:
try not to use a `_` separated prefix since it'll confuse how config uses env vars